packages/client/src/lib.js

```diff
@@ -35,8 +38,24 @@ const MAX_CHUNK_SIZE = 1024 * 1024 * 10 // chunk to ~10MB CARs
  * @typedef {import('./lib/interface.js').Pin} Pin
  * @typedef {import('./lib/interface.js').CarReader} CarReader
  * @typedef {import('ipfs-car/blockstore').Blockstore} BlockstoreI
+ * @typedef {import('./lib/interface.js').RateLimiter} RateLimiter
  */
 
+/**
+ * @returns {RateLimiter}
+ */
+export function createRateLimiter() {
+  const throttle = throttledQueue(RATE_LIMIT_REQUESTS, RATE_LIMIT_PERIOD)
+  return () => throttle(() => {})
+}
```

Comment:
Yeah the `throttled-queue` API is different, you pass your task as a function to it and it returns a promise that resolves when your task completes.

In the API I'm proposing here you just call the rate limiter function and it resolves when you can continue.

The code above just makes the throttle API work like I wanted it.

For context, I didn't really want an API that takes the task as a function, because you have to type it with generics and make sure you handle errors if the task throws. We can adopt the `throttle` API if you'd prefer - I don't have super strong feels about it.